omim2obo/main.py

```diff
@@ -230,19 +241,67 @@ def run(use_cache: bool = False):
         for mim_number in pheno_series[ps_id][1]:
             graph.add((OMIM[mim_number], RDFS.subClassOf, OMIMPS[ps_id]))
 
-    # Morbid map (cyto locations)
-    morbid_map: Dict[str, List[str]] = parse_morbid_map(
-        retrieve_mim_file('morbidmap.txt', download_files_tf))
-    for mim_number in morbid_map:
-        phenotype_mim_number, cyto_location = morbid_map[mim_number]
-        if phenotype_mim_number:
-            # RO:0003303 == 'causes condition' https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0003303
-            graph.add((OMIM[mim_number], RO['0003303'], OMIM[phenotype_mim_number]))
+    # Morbid map
+    morbid_map: Dict = parse_morbid_map(get_mim_file('morbidmap.txt', download_files_tf))
+    for mim_number, mim_data in morbid_map.items():
+        # todo?: unused `mim_data` keys. Should they be used?
+        #  - phenotype_label: Similar to p_lab in 'assocs', but has more info
+        #  - gene_symbols
+        cyto_location: str = mim_data['cyto_location']
         if cyto_location:
             # What's 9606chr - joeflack4 2021/11/11
             chr_id = '9606chr' + cyto_location
-            # What's RO['0002525'] - joeflack4 2021/11/11
+            # RO:0002525 (is subsequence of)
+            # https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0002525
             graph.add((OMIM[mim_number], RO['0002525'], CHR[chr_id]))
+        assocs: List[Dict] = mim_data['phenotype_associations']
+        print()
+        for assoc in assocs:
+            # p_lab currently not used
+            p_mim, p_lab, p_map_key, p_map_lab = \
+                assoc['phenotype_mim_number'], assoc['phenotype_label'], assoc['phenotype_mapping_info_key'], \
+                assoc['phenotype_mapping_info_label']
+            # Provenance: https://github.com/monarch-initiative/omim/issues/78
+            if not p_mim:
+                continue
+            # Provenance: https://github.com/monarch-initiative/omim/issues/79#issuecomment-1319408780
+            if p_map_key == '1':
+                continue
+
+            # Precalc: determine mapping predicate
+            # RO:0003302 (causes or contributes to condition)
+            # https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0003302
+            # Provenance for this decision:
+            # - Multiple rows, same mapping key: https://github.com/monarch-initiative/omim/issues/75
+            # - Multiple rows, diff mapping keys: https://github.com/monarch-initiative/omim/issues/81
+            predicate = RO['0003302']  # default if `len(assocs) > 1`
```

Comment:
@matentzn Just wanted to clarify I didn't misunderstand anything. See: https://github.com/monarch-initiative/omim/issues/75#issuecomment-1320995876